CONTRIBUTING.md

```diff
@@ -155,79 +102,126 @@ For example, in vscode your `.vscode/settings.json` should look like:
+  - If you approve of the PR, you are responsible for fixing any of the issues mentioned here and more.
+- If you sat down with the PR submitter and did a pairing review please note that in the `Approval`, or your PR comments.
+- If you are only making "surface level" reviews, submit any notes as `Comments` without adding a review.
 
-- ensure all github requirements pass
-- squash and merge pull request 
+### PR merge procedure
 
-### Release Procedure
+- Ensure all GitHub requirements pass.
+- Squash and merge pull request.
 
-- Start on `main`
-- Create the release candidate branch `rc/v*` (going forward known as **RC**)
-  and ensure it's protected against pushing from anyone except the release
-  manager/coordinator
-  - **no PRs targeting this branch should be merged unless exceptional circumstances arise**
-- On the `RC` branch, prepare a new version section in the `CHANGELOG.md`
-  - All links must be link-ified: `$ python ./scripts/linkify_changelog.py CHANGELOG.md`
-- Run external relayer tests against the prepared RC
-- If errors are found during the relayer testing, commit the fixes to `main`
-  and create a new `RC` branch (making sure to increment the `rcN`)
-- After relayer testing has successfully completed, create the release branch
-  (`release/vX.XX.X`) from the `RC` branch
-- Create a PR to `main` to incorporate the `CHANGELOG.md` updates
-- Tag the release (use `git tag -a`) and create a release in Github
-- Delete the `RC` branches
+## Branching model and release
 
-### Point Release Procedure
+User-facing repos should adhere to the [trunk based development branching model](https://trunkbaseddevelopment.com/).
 
-At the moment, only a single major release will be supported, so all point releases will be based
-off of that release.
+ibc-go utilizes [semantic versioning](https://semver.org/). Please our [`RELEASES.md`](./RELEASES.md) for more information.
+
+### New major release branch
+
+Pre-requisites for creating a release branch for a new major version:
+
+1. Bump [Go package version](https://github.com/cosmos/ibc-go/blob/main/go.mod#L3).
+2. Change all imports. For example: if the next major version is `v3`, then change all imports starting with `github.com/cosmos/ibc-go/v2` to `github.com/cosmos/ibc-go/v3`).
+
+Once the above pre-requisites are satified:
+
+1. Start on `main`.
+2. Create the release branch (`release/vX.XX.X`). For example: `release/v3.0.x`.
+
+### New minor release branch
+
+1. Start on the latest release branch in the same major release line. For example: the latest release branch in the `v3` release line is `v3.2.x`.
+2. Create branch from the release branch. For example: create branch `release/v3.3.x` from `v3.2.x`.
+
+Post-requisites for both new major and minor release branches:
+
+1. Add branch protection rules to new release branch.
+2. Add backport task to [`mergify.yml`](https://github.com/cosmos/ibc-go/blob/main/.github/mergify.yml).
+3. Create label for backport (e.g.`backport-to-v3.0.x`).
```

Comment:
Personally, I think we should move this information to some other document. It is only relevant to the folks who do releases. I think CONTRIBUTING.md should only contain information we expect all developers to know